keycloak_setup/institution_list.py

```diff
@@ -28,371 +30,619 @@ class InstitutionMeta(TypedDict):
-        "abbreviation": "OSU",
-        "is_US": True,
-        "region": _US_CAN,
+        "authorlist": True,
+        "authorlists": {
+            "astro": "Dept. of Astronomy, Ohio State University, Columbus, OH 43210, USA",
+            "physics": "Dept. of Physics and Center for Cosmology and Astro-Particle Physics, Ohio State University, Columbus, OH 43210, USA",
+        },
+        "_ldap_o": "osu",
     },
     "Oxford": {
         "cite": "Dept. of Physics, University of Oxford, Parks Road, Oxford OX1 3PU, UK",
         "abbreviation": "UOX",
         "is_US": False,
         "region": _EUROPE,
+        "authorlist": True,
+        "_ldap_o": "oxford",
     },
     "Padova": {
         "cite": "Dipartimento di Fisica e Astronomia Galileo Galilei, Università Degli Studi di Padova, 35122 Padova PD, Italy",
         "abbreviation": "UNIPD",
         "is_US": False,
         "region": _EUROPE,
+        "authorlist": True,
     },
-    "Penn-State-Astronomy": {
+    "Penn-State": {
         "cite": "Dept. of Astronomy and Astrophysics, Pennsylvania State University, University Park, PA 16802, USA",
         "abbreviation": "PSU",
         "is_US": True,
         "region": _US_CAN,
-    },
-    "Penn-State-Physics": {
-        "cite": "Dept. of Physics, Pennsylvania State University, University Park, PA 16802, USA",
-        "abbreviation": "PSU",
-        "is_US": True,
-        "region": _US_CAN,
+        "authorlist": True,
+        "authorlists": {
+            "astro": "Dept. of Astronomy and Astrophysics, Pennsylvania State University, University Park, PA 16802, USA",
+            "physics": "Dept. of Physics, Pennsylvania State University, University Park, PA 16802, USA",
+        },
+        "_ldap_o": "psu",
     },
     "Rochester": {
         "cite": "Dept. of Physics and Astronomy, University of Rochester, Rochester, NY 14627, USA",
         "abbreviation": "ROCHESTER",
         "is_US": True,
         "region": _US_CAN,
+        "authorlist": True,
+        "_ldap_o": "rochester",
     },
     "SD-Mines-Tech": {
         "cite": "Physics Department, South Dakota School of Mines and Technology, Rapid City, SD 57701, USA",
         "abbreviation": "SDSMT",
         "is_US": True,
         "region": _US_CAN,
+        "authorlist": True,
+        "_ldap_o": "sdsmt",
+    },
+    "Queens": {
+        "cite": "Dept. of Physics, Engineering Physics, and Astronomy, Queen's University, Kingston, ON K7L 3N6, Canada",
+        "abbreviation": "QUEEN'S",
+        "is_US": False,
+        "region": _US_CAN,
+        "authorlist": True,
+        "_ldap_o": "queensu",
     },
     "SNOLAB": {
         "cite": "SNOLAB, 1039 Regional Road 24, Creighton Mine 9, Lively, ON, Canada P3Y 1N2",
-        "abbreviation": "QUEEN'S",
+        "abbreviation": "SNOLAB",
         "is_US": False,
         "region": _US_CAN,
+        "authorlist": False,
```

Comment:
This is going to cause some breaks as is. I thought Queen's was SNOLAB for our purposes.

Reply:
So, apparently both members from SNOLAB moved to Queen's (they might have had a dual affiliation that ended), and I changed it in the author list here: https://wipacit.atlassian.net/browse/HELPDESK-3333

Reply:
Changing the abbreviation isn't ideal, but I can make a workaround (in the MOU Dash) since this was most likely a data inconsistency in the first place and should be fixed.

Reply:
"Queens" or "Queen's University" for the full name?

Reply:
I have no preference.

Reply:
How about adding the apostrophe at a minimum? Just googled it, "Queens" can refer to "Queens University of Charlotte" vs "Queen's University at Kingston".

Reply:
Note that the dict key isn't a great indicator of a name. It's a short-hand way to refer to an institution.  The "cite" is better as an official name, but it's not short.  If you want better names, maybe we should add a "name" attr?

Reply:
I'd also like to not add extra characters.  The dict keys have so far avoided spaces or apostrophes, which makes them nice to sort.

Reply:
I was treating the key as the official name. But I agree with you.

How about this:
- add a `"name"` field which is the public-facing name
- the `"abbreviation"` field is the publicly-facing shorthand name
- change the key to the new `"_o_ldap"` value (and get rid of `"_o_ldap"`)

Reply:
Note that "_o_ldap" won't appear as an attr in Keycloak, it's just for the LDAP import script.  And some of them don't make any sense (I had to look at other things to figure out which institution it was).  So not sure I'd want to use that as the key.

Reply:
I mention `"_o_ldap" ` because it's short and alpha-only. We could use a uuid. 

Do we need a key? Or can a list suffice?

Reply:
Either way, let's add a `"name"` field.

Reply:
I think the thing I was using the key for was the Keycloak group name.  Which has some limitations.

I might need to add "city" as well, to maintain the author list sorting (because someone way back decided that's how it was sorted).

Reply:
It sounds like every service wants to refer to an institution in a different way, yeah? Can we just make this a list, and move the current key to a new field, `"keycloak_group_name"`?

Reply:
(in slack we settled on keeping this as a dict keyed with the keycloak group name string since this will guarantee uniqueness among the collection.)

Reply:
linking to https://github.com/WIPACrepo/mou-dashboard/issues/47 for future reference